include/bactria/Incident.hpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2021 Jan Stephan
+ *
+ * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by
+ * the European Commission - subsequent versions of the EUPL (the “Licence”).
+ * You may not use this work except in compliance with the Licence.
+ * You may obtain a copy of the Licence at:
+ *
+ *     http://ec.europa.eu/idabc/eupl.html
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the Licence is distributed on an “AS IS” basis, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  Licence permissions and limitations under the Licence.
+ */
+
+#ifndef BACTRIA_INCIDENT_HPP
+#define BACTRIA_INCIDENT_HPP
+
+#   include <string>
+#   include <type_traits>
+#   include <utility>
+
+namespace bactria
+{
+    template <typename... TIncidents>
+    class Report;
+
+    template <typename TValue>
+    class Incident
+    {
+        static_assert(std::is_fundamental<TValue>::value || std::is_same<TValue, std::string>::value ||
```

Comment:
So my innocent `enum : int32_t { good, bad, maybe_ok; }` is not supported? :/

Reply:
Not in the current version. This might change in the future based on user feedback and backend capabilities.